src/Features/Core/Portable/Completion/Providers/AbstractObjectInitializerCompletionProvider.cs

```diff
@@ -96,5 +88,42 @@ protected virtual bool IsInitializable(ISymbol member, INamedTypeSymbol containi
                 member.MatchesKind(SymbolKind.Field, SymbolKind.Property) &&
                 member.IsAccessibleWithin(containingType);
         }
+
+        private static bool CanSupportObjectInitializer(ISymbol symbol, ISymbol within)
+        {
+            if (symbol is IPropertySymbol propertySymbol)
+            {
+                return propertySymbol.GetMethod != null && !propertySymbol.Type.IsStructType() && HasAccesseblePropertiesOrFields(propertySymbol.Type, within);
+            }
+
+            return false;
+        }
+
+        private static bool HasAccesseblePropertiesOrFields(ITypeSymbol type, ISymbol within)
```

Comment:
Since it also checks for CanSupportCollectionInitializer, this should be named HasObjectInitializableMembers() or similar.

Reply:
📝 Accessibility does not need to be checked to determine whether the syntax is allowed.